### PR TITLE
Memory Footprint Resolved

### LINKED ISSUE #522: Minimizing Memory Footprint for JSON Lines Files
#### Example

```python
table = etl.fromjson('path_to_file.jsonl', lines=True)
print(etl.nrows(table))
```

#### Problem Description

Upon ingesting a delimited JSON (.jsonl) file, the memory footprint is huge. For a 577M file, it takes roughly 2.87G of RAM and it there seems to be a visible improvement.

#### Version

* Latest (1.6.5) version of `petl`
* Python 3.7.x
* Mac
* Using pip into a virtualenv

#### Proposal

Rather iterating through the JSON in each line and appending the dict somewhere, which takes huge chunk of memory, we can directly per line dict into tuples.

#### Progress

I already have tested it out and my new implementation is roughly taking 30M memory to ingest 577M JSON lines file (uncompressed). I think it's ready to be implemented after doing a small technical spike.

### PR DESCRIPTION
**Pull request by fahadsiddiqui**
Resolves #522 

## Changes
1. Memory footprint reduced

## Checklist

* [ ] Code
  * [ ] Includes unit tests
  * [ ] New functions have docstrings with examples that can be run with doctest
  * [ ] New functions are included in API docs
  * [ ] Docstrings include notes for any changes to API or behaviour
  * [x] All changes documented in docs/changes.rst
* [ ] Testing
  * [ ] \(Optional) Tested local against remote servers
  * [x] Travis CI passes (unit tests run under Linux)
  * [x] AppVeyor CI passes (unit tests run under Windows)
  * [ ] Unit test coverage has not decreased (see Coveralls)
* [ ] Changes
  * [ ] \(Optional) Just a proof of concept
  * [ ] \(Optional) Work in progress
  * [x] Ready to review
  * [x] Ready to merge

### CODE REVIEW

**Comment by fahadsiddiqui** [comment]
@juarezr @alimanfoo please read #522 on why I did this enhancement. It is a huge improvement to my previous implementation. Processing a 577M file took around 2.87G of RAM (may be keeping dicts appended where the dicts were huge). After making this change, the memory footprint is decreast to just 26M for a 577M of JSON lines file. Huge improvement!

File: petl/io/json.py [hunk 1]
```diff
@@ -173,6 +171,22 @@ def __iter__(self):
         return iterdicts(self.dicts, self.header, self.sample, self.missing)
 
 
+def iterjlines(f, header, missing):
+    it = iter(f)
+
+    if header is None:
+        header = list()
+        peek, it = iterpeek(it, 1)
```
Comment (fahadsiddiqui) [inline petl/io/json.py:179]: JSON lines file will have complete schema on first line or any other line, so we can safely peek just one line.

**Review by juarezr — APPROVED** [review]
Nice work!

**Comment by juarezr** [comment]
Do you plan more improvements or fixes?

**Pull request merged**